### PR TITLE
Add predicu to icubam

### PR DESCRIPTION
**Pull request by tgy**

### CODE REVIEW

File: icubam/predicu/tests/test_load_public_data.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import datetime
+
+import predicu.data
+
+
+def test_load_public_data():
+  data = predicu.data.load_public_data()
```
Comment (rth) [inline icubam/predicu/tests/test_load_public_data.py:7]: I'm wondering if we should skip this by default. And have some pytest flag to run it optionally.

File: icubam/predicu/__main__.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    logging.info("creating directory %s" % output_dir)
+    os.makedirs(output_dir)
+  logging.info("exporting data to %s" % output_dir)
+  if not os.path.isdir(output_dir):
+    raise IOError(f"Output dir `{output_dir}' does not exist.")
+  datetimestr = datetime.datetime.now().strftime("%Y-%m-%d_%Hh%M")
+  filename = "predicu_data_clean_{}.csv".format(datetimestr)
+  path = os.path.join(output_dir, filename)
+  d = predicu.data.load_all_data(
+    clean=True, cache=False, api_key=args.api_key, max_date=args.max_date
```
[the remark this answers is not in the document]
Comment (tgy) [reply icubam/predicu/__main__.py:21]: yep, let's merge this as a base and then move on with other PRs. easier to track!

File: icubam/predicu/__main__.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+if __name__ == "__main__":
+  logging.getLogger().setLevel(logging.INFO)
+  parser = argparse.ArgumentParser()
+  subparsers = parser.add_subparsers()
+  parser_export = subparsers.add_parser("export", help="export data")
+  parser_export.set_defaults(func=export_data)
+  parser_export.add_argument("--output-dir", "-o", default="/tmp/")
+  parser_export.add_argument("--api-key", default=None)
+  parser_export.add_argument(
```
[the remark this answers is not in the document]
Comment (tgy) [reply icubam/predicu/__main__.py:35]: agreed. could you open an issue?

File: icubam/predicu/data.py [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+
+def load_all_data(
+  clean=True,
+  cache=False,
+  spread_cum_jump_correction=False,
+  api_key=None,
+  max_date=None,
+  icubam_data: pd.DataFrame = None,
+):
+  if cache and os.path.isfile(DATA_PATHS["icubam_cache"]):
```
[the remark this answers is not in the document]
Comment (tgy) [reply icubam/predicu/data.py:65]: i'll remove caching in this PR.

File: icubam/predicu/data.py [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+  d = d.sort_values(by=["date", "icu_name"])
+  if max_date is not None:
+    logging.info("data loaded's max date will be %s (excluded)" % max_date)
+    d = d.loc[d.date < pd.to_datetime(max_date).date()]
+  if cache and clean:
+    d.to_hdf(DATA_PATHS["icubam_cache"], "values")
+  return d
+
+
+def load_icubam_data(api_key):
```
[the remark this answers is not in the document]
Comment (tgy) [reply icubam/predicu/data.py]: done

File: icubam/predicu/tests/test_load_public_data.py (hunk 1 above)
Comment (tgy) [reply icubam/predicu/tests/test_load_public_data.py:7]: yep let's do that!

**Comment by rth** [comment]
Do we really need pytables? HDF5 is a heavier dependency. It wasn't used before, maybe keep it optional?

**Pull request merged**